geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala

```diff
@@ -45,24 +38,24 @@ class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLo
     val sft = queryPlanner.sft
 
     // pull out any dates from the filter to help narrow down the attribute ranges
-    val dates = {
-      val (dateFilters, _) = filter.secondary.map {
-        case a: And => partitionPrimaryTemporals(a.getChildren, queryPlanner.sft)
-        case f      => partitionPrimaryTemporals(Seq(f), queryPlanner.sft)
-      }.getOrElse((Seq.empty, Seq.empty))
-      val interval = extractInterval(dateFilters, queryPlanner.sft.getDtgField)
-      if (interval == everywhen) None else Some((interval.getStartMillis, interval.getEndMillis))
+    val dates = for {
+      dtgField  <- sft.getDtgField
+      secondary <- filter.secondary
+      intervals = FilterHelper.extractIntervals(secondary, dtgField)
+      if intervals.nonEmpty
+    } yield {
+      (intervals.map(_._1.getMillis).min, intervals.map(_._2.getMillis).max)
     }
 
     // for an attribute query, the primary filters are considered an OR
     // (an AND would never match unless the attribute is a list...)
-    val propsAndRanges = filter.primary.map(getPropertyAndRange(queryPlanner.sft, _, dates))
-    val attributeSftIndex = propsAndRanges.head._1
-    val ranges = propsAndRanges.map(_._2)
+    val propsAndRanges = getBounds(sft, filter.singlePrimary.get, dates)
+    val attribute = propsAndRanges.head.attribute
```

Comment:
Similar to above - we wouldn't get to this code if it was empty. But I'll add a headOption.getOrElse to fail more obviously.